plugins/search/src/components/SearchResult/SearchResult.tsx

```diff
@@ -0,0 +1,230 @@
+
+import SearchApi from '../../apis';
+
+const useStyles = makeStyles(() => ({
+  searchTerm: {
+    background: '#eee',
+    padding: '0.2%',
+    borderRadius: '10%',
+  },
+  tableHeader: {
+    marginTop: '10px',
+    display: 'flex',
+  },
+  divider: {
+    width: '1px',
+    margin: '0 20px',
+    padding: '15px 0',
+  },
+}));
+
+type SearchResultProps = {
+  currentTarget?: string;
+};
+
+type TableHeaderProps = {
+  searchTerm?: string;
+  numberOfSelectedFilters: number;
+  numberOfResults: number;
+  handleToggleFilters: () => any;
+};
+
+type Filters = {
+  selected: string;
+  checked: Array<string | null>;
+};
+
+// TODO: move out column to make the search result component more generic
+const columns: TableColumn[] = [
+  {
+    title: 'Component Id',
+    field: 'name',
+    highlight: true,
+  },
+  {
+    title: 'Description',
+    field: 'description',
+  },
+  {
+    title: 'Owner',
+    field: 'owner',
+  },
+  {
+    title: 'Kind',
+    field: 'kind',
+  },
+  {
+    title: 'LifeCycle',
+    field: 'lifecycle',
+  },
+];
+
+const TableHeader = ({
+  searchTerm,
+  numberOfSelectedFilters,
+  numberOfResults,
+  handleToggleFilters,
+}: TableHeaderProps) => {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.tableHeader}>
+      <FilterButton
+        numberOfSelectedFilters={numberOfSelectedFilters}
+        handleToggleFilters={handleToggleFilters}
+      />
+      <Divider className={classes.divider} orientation="vertical" />
+      <Grid item sm={12}>
+        {searchTerm ? (
+          <Typography variant="h6">
+            {`${numberOfResults} `}
+            {numberOfResults > 1 ? `results for ` : `result for `}
+            <span className={classes.searchTerm}>"{searchTerm}"</span>{' '}
+          </Typography>
+        ) : (
+          <Typography variant="h6">{`${numberOfResults} results`}</Typography>
+        )}
+      </Grid>
+    </div>
+  );
+};
+
+const SearchResult = ({ currentTarget }: SearchResultProps) => {
+  const [showFilters, toggleFilters] = useState(false);
+  const [filters, setFilters] = useState<Filters>({
+    selected: 'All',
+    checked: [],
+  });
+
+  const [filteredResults, setFilteredResults] = useState<Array<object>>([]);
+
+  const searchApi = new SearchApi();
+
+  const { loading, error, value: results } = useAsync(() => {
+    return searchApi.getSearchData();
+  }, []);
+
+  useEffect(() => {
+    if (results) {
+      let withFilters = results;
+
+      // apply filters
+
+      // filter on selected
+      if (filters.selected !== 'All') {
+        withFilters = results.filter((result: any) =>
+          filters.selected.includes(result.kind),
+        );
+      }
+
+      // filter on checked
+      if (filters.checked.length > 0) {
+        withFilters = withFilters.filter((result: any) =>
+          filters.checked.includes(result.lifecycle),
+        );
+      }
+
+      // filter on searchTerm
+      if (currentTarget) {
```

Comment:
could rename this to `searchTerm` or maybe better `filterText` to make its purpose a bit clearer (a "term" feels like a single word to me)